libopenage/gamestate/game_main.cpp

```diff
@@ -37,17 +38,14 @@ GameMain::GameMain(const Generator &generator)
 }
 
 GameMain::~GameMain() {
-	for (auto &p : this->players) {
-		delete p;
-	}
 }
 
 unsigned int GameMain::player_count() const {
 	return this->players.size();
 }
 
 Player *GameMain::get_player(unsigned int player_id) {
-	return this->players.at(player_id);
+	return this->players.at(player_id).get();
```

Comment:
ah yes - I was on master! 
on second thought - if the player is never stored anywhere (like in a class variable) there should be no trouble using ```shared_ptr::get```. The only place where this might be useful is to store the owning player of each unit - and avoiding circular references is not something you want to waste your brainpower on. 

I agree with your solution!